### DISCUSSION TITLE
آپدیت فرانت اند قالب خبرنامه

### DISCUSSION THREAD

**Discussion by mhdi-nzari** [softwaretalks/newsletter#566] [Ideas]
سلام امیدوارم حالتون خوب باشه 
من من فرانت اند کار میکنم و دوست دارم قالب خبرنامه رو فرانتش رو بزنم میتونم کمک کنم به اینکار ؟

**Comment by Hameds** [comment]
سلام و وقت بخیر 

ممنونم از شما. بله حتما، باعث افتخاره. پیش از اجرا اگر امکان داشت طراحی رو در ادامه همین گفتگو ارسال بفرمایید.

**Reply by mhdi-nzari** [reply]
فقط بهم میگین چطوری باید قالب رو دریافت کنم و تغییرش بدم 
از کدوم قسمت

**Reply by Hameds** [reply]
ما دو تا قالب مشابه هم داریم (dark , light) که هر دو از آدرس زیر قابل مشاهده هستند. کدی که امیرحسین نوشته، ایشوهای لازم رو در این قالب‌ها قرار می‌ده و یک html نهایی می‌سازه که از طریق API به سرویس ارسال ایمیل خبرنامه ارسال می‌شه.

https://github.com/softwaretalks/newsletter/tree/main/src/EMAIL_TEMPLATES

**Comment by mhdi-nzari** [comment]
محتوا آماده هست برای پوش کردن میگه دسترسی نداری 
mahdi-nazari.gb008@gmail.com

**Reply by Hameds** [reply]
خیلی ممنونم. اگر ممکنه یک fork از پروژه ایجاد و در قالب Pull Request ارسال بفرمایید.

**Comment by mhdi-nzari** [comment]
انجام شد.

**Reply by mhdi-nzari** [reply]
مشکلی چیزی نداره و همچی مرتبه ؟

**Reply by Hameds** [reply]
ممنونم 👍  بررسی می‌کنم.

**Reply by ohmydevops** [reply]
نکته ای که وجود داره اینه که الان چطوری میتونیم از این پروژه فرانتی به عنوان یک تمپلیت استفاده کنیم؟
مثلا اینجا رو نگاه کنید: 
https://github.com/mhdi-nzari/newsletter/blob/7d419ac2c40de3d92466dd7544f8ffbe9cd21eee/src/EMAIL_TEMPLATES/newsletter.html#L54

ما در نسخه فعلی با php داده‌ها رو میچوپونیم توی تمپلیت و فایل html نهایی رو میسازیم. در حال حاضر با پروژه شما چطور میتونیم این کار رو انجام بدیم؟ ایا برای این موضوع که از بیرون بشه بهش داده inject کنیم سازوکاری وجود داره؟

**Reply by ohmydevops** [reply]
برداشت من این هست که روالش این طوری میشه:

- ابتدا داده‌ها رو به هر شکلی که شده در فایل html بچوپونیم
- با npm build پروژه رو بیلد کنیم و خروجی نهایی رو در ایمیل ارسال کنیم (دقت بشه که خروجی نهایی باید در یک فایل باشه یعنی یک فایل html که همه چیزهاش داخلشه و مثلا فایل css js خارجی نداشته باشیم)

**Reply by Hameds** [reply]
درسته. البته ما در نهایت باید سورس خروجی نهایی html رو برای API جهت ارسال بفرستیم و فکر کنم خود فایل خروجی html رو نمی‌شه
